app/controllers/shopify_app/callback_controller.rb

```diff
@@ -6,38 +6,57 @@ class CallbackController < ActionController::Base
     include ShopifyApp::LoginProtection
 
     def callback
-      unless auth_hash
-        return respond_with_error
-      end
+      return respond_with_error if invalid_request?
+
+      store_access_token_and_build_session
 
-      if jwt_request? && !valid_jwt_auth?
-        Rails.logger.debug("[ShopifyApp::CallbackController] Invalid JWT auth detected.")
-        return respond_with_error
+      if start_user_token_flow?
+        return respond_with_user_token_flow
       end
 
+      perform_post_authenticate_jobs
+
+      respond_successfully
+    end
+
+    private
+
+    def respond_successfully
       if jwt_request?
-        Rails.logger.debug("[ShopifyApp::CallbackController] JWT request detected. Setting shopify session...")
```

Comment:
Are we intentionally not replacing the debug logs?

Reply:
Just added the logs back